### PR TITLE
sync: Add metrics

### LINKED ISSUE #1153: Add metrics for sync status
The `best_known_block_number` and `blockchain_height` implemented in the [Nethermind client](https://docs.nethermind.io/nethermind/ethereum-client/metrics/blockchain) look particularly useful.

### PR DESCRIPTION
**Pull request by Exca-DK**
closes #1153

### CODE REVIEW

File: sync/sync.go [hunk 1]
```diff
@@ -41,8 +41,12 @@ type Synchronizer struct {
...
 
 	// metrics
-	opTimers    *prometheus.HistogramVec
-	totalBlocks prometheus.Counter
+	opTimersMetric     *prometheus.HistogramVec
+	totalBlocksMetric  prometheus.Counter
+	heightBlockMetric  prometheus.Gauge
+	bestBlockMetric    prometheus.Gauge
+	reorgMetric        prometheus.Counter
+	transactionsMetric prometheus.Counter
```
Comment (joshklop) [inline sync/sync.go:49]: nit: maybe we can adjust the naming pattern for clarity. Example:

```go
blockCount prometheus.Counter
chainHeightGauge prometheus.Gauge
reorgCount prometheus.Counter
transactionCount prometheus.Counter
// etc.
```

File: sync/sync.go [hunk 2]
```diff
@@ -204,7 +231,6 @@ func (s *Synchronizer) verifierTask(ctx context.Context, block *core.Block, stat
 					s.catchUpMode = isBehind
 				}
 			}
-
```
Comment (joshklop) [inline sync/sync.go:207]: nit: spurious diff

**Review by joshklop — COMMENTED** [review]
Probably isn't good for this PR, but it would be nice to add tests for metrics imho.

Maybe with #1168

**Review by joshklop — APPROVED** [review]
> happy uk linter

:joy:

File: sync/sync.go [hunk 3]
```diff
@@ -54,16 +58,39 @@ func New(bc *blockchain.Blockchain, starkNetData starknetdata.StarknetData,
...
+			Namespace: "sync",
+			Name:      "blockchain_height",
+		}),
+		bestBlockGauge: prometheus.NewGauge(prometheus.GaugeOpts{
+			Namespace: "sync",
+			Name:      "best_known_block_number",
+		}),
+		reorgCount: prometheus.NewCounter(prometheus.CounterOpts{
+			Namespace: "sync",
+			Name:      "reorganisations",
```
Comment (Exca-DK) [inline sync/sync.go:79]: Although I did change it to satisfy uk based linter, this was based on [nethermind_reorganizations](https://docs.nethermind.io/nethermind/ethereum-client/metrics/blockchain)

Comment (joshklop) [reply sync/sync.go:79]: That's ok. We use UK spelling elsewhere.

**Pull request merged**